### PR TITLE
feat(slab synchronizer): Add slab thickness synchronizer

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

Adds a slab thickness synchronizer

### Testing

yarn example crosshairs

Run/test with the synchronizer on/off

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Review by vitalii-komenda — APPROVED** [review]
We need this feature

File: packages/tools/src/synchronizers/callbacks/slabThicknessSyncCallback.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  const slabThickness = (
+    sViewport as Types.IVolumeViewport
+  ).getSlabThickness?.();
+  if (!slabThickness) {
+    return;
+  }
+  (tViewport as Types.IVolumeViewport).setSlabThickness?.(slabThickness);
+  tViewport.render();
+}
```
Comment (sedghi) [inline packages/tools/src/synchronizers/callbacks/slabThicknessSyncCallback.ts:29]: didn't we decide to do a properties synchronizer?

File: packages/tools/src/store/SynchronizerManager/Synchronizer.ts [hunk 2]
```diff
@@ -66,6 +66,13 @@ class Synchronizer {
     this._viewportOptions[viewportId] = options;
   }
 
+  /**
+   * Sets a synchronizer enabled
+   */
+  public setEnabled(enabled: boolean) {
```
Comment (sedghi) [inline packages/tools/src/store/SynchronizerManager/Synchronizer.ts:72]: can we add a setDisabled too?

File: packages/tools/examples/crossHairs/index.ts [hunk 3]
```diff
@@ -190,6 +199,28 @@ addDropdownToToolbar({
...
+  synchronizer = createSlabThicknessSynchronizer(synchronizerId);
+
+  // Add viewports to VOI synchronizers
+  [viewportId1, viewportId2, viewportId3].forEach((viewportId) => {
+    synchronizer.add({
+      renderingEngineId,
+      viewportId,
+    });
+  });
+  synchronizer.setEnabled(false);
```
Comment (sedghi) [inline packages/tools/examples/crossHairs/index.ts:221]: please add a comment that this is just for the purpose of this demo, in reality you want it on

**Review by sedghi — COMMENTED** [review]
see my comments thanks

File: packages/tools/src/synchronizers/callbacks/slabThicknessSyncCallback.ts (hunk 1 above)
Comment (wayfarer3130) [reply packages/tools/src/synchronizers/callbacks/slabThicknessSyncCallback.ts:29]: I'm working on a view position synchronizer, but I think the slab thickness stands on it's own as something that one would turn on/off by itself.

File: packages/tools/src/store/SynchronizerManager/Synchronizer.ts (hunk 2 above)
Comment (wayfarer3130) [reply packages/tools/src/store/SynchronizerManager/Synchronizer.ts:72]: That isn't normally how one does it - just call setEnabled(false)

**Review by sedghi — APPROVED** [review]

**Pull request merged**